src/controllers/GameController.ts

```diff
@@ -28,7 +32,13 @@ class GameController {
             if (isLegalMove) {
                 this.gameEngine.movePiece(this.activeSquare, square);
                 this.boardView.render(this.gameEngine.board);
-                this.changePlayer();
+
+                const isPawnPromotion = this.gameEngine.checkPawnPromotion(square);
```

Comment:
zmienna to boolean, nazwa powinna to wskazywać, może coś w stylu is(Pawn)PromotionPossible? bo chyba nie ma takiego tworu jak isPawnPromotable :P